rust/hyperlane-base/src/db/rocks/hyperlane_db.rs

```diff
@@ -320,3 +271,49 @@ where
         Ok(result)
     }
 }
+
+/// Generate a call to ChainSetup for the given builder
+macro_rules! build_store_and_retrieve {
```

Comment:
Oh, also instead of `build_store_and_retrieve` I think `make_store_and_retrieve` would be better. Builder implies a runtime builder whereas this is a compile-time thing.